backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java

```diff
@@ -212,15 +212,47 @@ public LiveOktaRepository(
           "Cannot update Okta user with unrecognized username");
     }
     User user = users.single();
+    updateUser(user, userIdentity);
+
+    return getOrganizationRoleClaimsForUser(user);
+  }
+
+  private void updateUser(User user, IdentityAttributes userIdentity) {
     user.getProfile().setFirstName(userIdentity.getFirstName());
     user.getProfile().setMiddleName(userIdentity.getMiddleName());
     user.getProfile().setLastName(userIdentity.getLastName());
     // Is it our fault we don't accommodate honorific suffix? Or Okta's fault they
     // don't have regular suffix? You decide.
     user.getProfile().setHonorificSuffix(userIdentity.getSuffix());
     user.update();
+  }
 
-    return getOrganizationRoleClaimsForUser(user);
+  public void reprovisionUser(IdentityAttributes userIdentity) {
+    UserList users = _client.listUsers(userIdentity.getUsername(), null, null, null, null);
```

Comment:
(This is definitely out of scope for this change, but I've been spending so much time looking at the Okta API that I thought I'd mention it)
Is the `username` the same as the user's login? Okta has an API call to fetch the user with login, but it doesn't seem to be available in the SDK/as a function of `Client`. If the username is the login, it may be worth considering fetching the user directly via the API instead of listing the users; it might help our Okta quota.
https://developer.okta.com/docs/reference/api/users/#response-example-11

Reply:
To clarify what I'm thinking a bit further, that API call returns the user's id (among other things) and we could then use that id to call `_client.getUser(userId)`. Then all our user logic would continue from there.

Reply:
Ahh, interesting!  That's great to know about.  I don't think the way it is done right now causes an extra api call to Okta, but it is much less clear what is happening compared to just fetching with the email address.

Yes, for us, `username` is the user's login.